### PR TITLE
Allow undeprecation of storages

### LINKED ISSUE #4548: Allow to undeprecate a storage
**Motivation**
Users may change their mind and want to use a storage again

**Acceptance criteria**
1. A new endpoint is provided to undeprecate storages

### PR DESCRIPTION
**Pull request by olivergrabinski**
Closes #4548

Screenshot 2023-12-05 at 14 47 28

Screenshot 2023-12-05 at 14 48 26

### CODE REVIEW

File: docs/src/main/paradox/docs/delta/api/storages-api.md [hunk 1]
```diff
@@ -37,23 +39,34 @@ local file-system structure and that Nexus has read and write access to the targ
   "readPermission": "{read_permission}",
   "writePermission": "{write_permission}",
   "capacity": "{capacity}",
-  "maxFileSize": {max_file_size}
+  "maxFileSize": {
```
Comment (imsdu) [inline docs/src/main/paradox/docs/delta/api/storages-api.md:42]: Is this on purpose ?

File: tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/StorageSpec.scala [hunk 2]
```diff
@@ -419,6 +419,25 @@ abstract class StorageSpec extends BaseIntegrationSpec {
     }
   }
 
+  "Undeprecating a storage" should {
+
+    "allow uploading a file again" in {
+      deltaClient
```
Comment (imsdu) [inline tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/StorageSpec.scala:425]: You can put those in a for comprehension

File: docs/src/main/paradox/docs/delta/api/storages-api.md (hunk 1 above)
Comment (olivergrabinski) [reply docs/src/main/paradox/docs/delta/api/storages-api.md]: Reformatted the whole file by mistake; diff on the file is legible now

File: tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/StorageSpec.scala (hunk 2 above)
Comment (olivergrabinski) [reply tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/StorageSpec.scala]: Done

**Review by imsdu — APPROVED** [review]

**Pull request merged**